guides/customizing_saving_and_serialization.py

```diff
@@ -0,0 +1,328 @@
+"""
+Title: Customizing Saving and Serialization
+Author: Neel Kovelamudi
+Date created: 2023/03/15
+Last modified: 2023/03/15
+Description: A more advanced guide on customizing saving for your layers and models.
+"""
+
+"""
+## Introduction
+
+This guide covers advanced methods that can be customized in Keras saving. For most
+users, the methods outlined in the primary
+[Serialize, save, and export guide](https://keras.io/guides/serialization_and_saving)
+are sufficient.
+"""
+
+"""
+### APIs
+We will cover the following APIs:
```

Comment:
When restoring a model, what order do these get executed in?

Reply:
When restoring a model, these get executed in the following order:
1) build from config
2) compile from config
3) load variables
4) load assets

Reply:
Cool, could we mention that somewhere?

Reply:
Sure! Added it to the APIs section.